templates/jail.local.j2

```diff
@@ -0,0 +1,205 @@
+fail2ban_agent = Fail2Ban/%(fail2ban_version)s
+
+#
+# Action shortcuts. To be used to define action parameter
+
+# Default banning action (e.g. iptables, iptables-new,
+# iptables-multiport, shorewall, etc) It is used to define
+# action_* variables. Can be overridden globally or per
+# section within jail.local file
+banaction = iptables-multiport
+banaction_allports = iptables-allports
+
+# The simplest action to take: ban only
+action_ = %(banaction)s[name=%(__name__)s, bantime="%(bantime)s", port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]
+
+# ban & send an e-mail with whois report to the destemail.
+action_mw = %(banaction)s[name=%(__name__)s, bantime="%(bantime)s", port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]
+            %(mta)s-whois[name=%(__name__)s, sender="%(sender)s", dest="%(destemail)s", protocol="%(protocol)s", chain="%(chain)s"]
+
+# ban & send an e-mail with whois report and relevant log lines
+# to the destemail.
+action_mwl = %(banaction)s[name=%(__name__)s, bantime="%(bantime)s", port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]
+             %(mta)s-whois-lines[name=%(__name__)s, sender="%(sender)s", dest="%(destemail)s", logpath=%(logpath)s, chain="%(chain)s"]
+
+# See the IMPORTANT note in action.d/xarf-login-attack for when to use this action
+#
+# ban & send a xarf e-mail to abuse contact of IP address and include relevant log lines
+# to the destemail.
+action_xarf = %(banaction)s[name=%(__name__)s, bantime="%(bantime)s", port="%(port)s", protocol="%(protocol)s", chain="%(chain)s"]
+             xarf-login-attack[service=%(__name__)s, sender="%(sender)s", logpath=%(logpath)s, port="%(port)s"]
+
+# ban IP on CloudFlare & send an e-mail with whois report and relevant log lines
+# to the destemail.
+action_cf_mwl = cloudflare[cfuser="%(cfemail)s", cftoken="%(cfapikey)s"]
+                %(mta)s-whois-lines[name=%(__name__)s, sender="%(sender)s", dest="%(destemail)s", logpath=%(logpath)s, chain="%(chain)s"]
+
+# Report block via blocklist.de fail2ban reporting service API
+# 
+# See the IMPORTANT note in action.d/blocklist_de.conf for when to
+# use this action. Create a file jail.d/blocklist_de.local containing
+# [Init]
+# blocklist_de_apikey = {api key from registration]
+#
+action_blocklist_de  = blocklist_de[email="%(sender)s", service=%(filter)s, apikey="%(blocklist_de_apikey)s", agent="%(fail2ban_agent)s"]
+
+# Report ban via badips.com, and use as blacklist
+#
+# See BadIPsAction docstring in config/action.d/badips.py for
+# documentation for this action.
+#
+# NOTE: This action relies on banaction being present on start and therefore
+# should be last action defined for a jail.
+#
+action_badips = badips.py[category="%(__name__)s", banaction="%(banaction)s", agent="%(fail2ban_agent)s"]
+#
+# Report ban via badips.com (uses action.d/badips.conf for reporting only)
+#
+action_badips_report = badips[category="%(__name__)s", agent="%(fail2ban_agent)s"]
+
```

Comment:
I think it would be useful to be able to configure filters and actions using Ansible variables, but that may be work for another pull request.

Reply:
Yes I thought about that too. But I guess it's better to add it later on in a sperate pull request.

Reply:
I mean creating new filters and actions in a dynamic way, otherwise we have to create a pull request in this role for every new filter/action.
We could an Ansible dict for this, which has the name of the action/filter as a key and a multi-line string as content. The content could then be written into the fail2ban config or into separate files.

Reply:
To clarify: the statement
```
action_badips_report = badips[category="%(__name__)s", agent="%(fail2ban_agent)s"]
```
defines a new action `action_badips_report` based on another action `badips` defined in `action.d/badips.conf`. It only sets two variables. Right now redefining actions in such a way isn't possible, but I guess we won't need such a feature anytime soon. It's possible to add that later.

Right now it's possible to specify the actions used in jails. For example a host could be configured like this:

```
fail2ban_jails:
  wordpress:
    filter: wordpress
    action: nginx
    …
```

The `wordpress` jail will use the filter `filter.d/wordpress.conf` and the action `action.d/nginx.conf`. So it's possible to specify what filter/action is used by a jail. However it's not possible atm to add custom actions/filters yet, but I'm going to add such a feature later on in a seperate pull request. It's basically nothing more than a file upload feature.

It might be useful to know that fail2ban already comes with a lot of filters and actions on default (e.g. an action for iptables). So I guess that we don't need to add custom filters and actions that often.